### ISSUE TITLE
Protractor testing

### ISSUE THREAD

**Issue by mchapman** [angular/material-start#30]
Can anyone get the tests to work?  I am getting

```
Starting selenium standalone server...
[launcher] Running 1 instances of WebDriver
Selenium standalone server started at http://192.168.0.11:38540/wd/hub
FFF

Failures:

  1) my app should automatically redirect to /view1 when location hash/fragment is empty
   Message:
     Error while waiting for Protractor to sync with the page: "angular could not be found on the window"
   Stacktrace:
     undefined

  2) my app view1 should render view1 when user navigates to /view1
   Message:
     Error: Index out of bound. Trying to access element at index: 0, but there are only 0 elements that match locator By.cssSelector("[ng-view] p")
   Stacktrace:
     Error: Index out of bound. Trying to access element at index: 0, but there are only 0 elements that match locator By.cssSelector("[ng-view] p")
    at Array.forEach (native)
Error
    at [object Object].<anonymous> (/home/mark/Projects/material-start/e2e-tests/scenarios.js:22:57)
From: Task: Asynchronous test function: it()
Error
    at [object Object].<anonymous> (/home/mark/Projects/material-start/e2e-tests/scenarios.js:21:5)
    at [object Object].<anonymous> (/home/mark/Projects/material-start/e2e-tests/scenarios.js:14:3)
    at Object.<anonymous> (/home/mark/Projects/material-start/e2e-tests/scenarios.js:5:1)

  3) my app view2 should render view2 when user navigates to /view2
   Message:
     Error: Index out of bound. Trying to access element at index: 0, but there are only 0 elements that match locator By.cssSelector("[ng-view] p")
   Stacktrace:
     Error: Index out of bound. Trying to access element at index: 0, but there are only 0 elements that match locator By.cssSelector("[ng-view] p")
    at Array.forEach (native)
Error
    at [object Object].<anonymous> (/home/mark/Projects/material-start/e2e-tests/scenarios.js:37:57)
From: Task: Asynchronous test function: it()
Error
    at [object Object].<anonymous> (/home/mark/Projects/material-start/e2e-tests/scenarios.js:36:5)
    at [object Object].<anonymous> (/home/mark/Projects/material-start/e2e-tests/scenarios.js:29:3)
    at Object.<anonymous> (/home/mark/Projects/material-start/e2e-tests/scenarios.js:5:1)

Finished in 1.299 seconds
3 tests, 3 assertions, 3 failures

Shutting down selenium standalone server.

/usr/local/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/atoms/error.js:113
  var template = new Error(this.message);
                 ^
UnknownError: unknown error: [ng:btstrpd] App Already Bootstrapped with this Element '&lt;body ng-app="starterApp" layout="column" ng-controller="UserController as ul" class="ng-scope"&gt;'
http://errors.angularjs.org/1.3.15/ng/btstrpd?p0=%26lt%3Bbody%20ng-app%3D%22starterApp%22%20layout%3D%22column%22%20ng-controller%3D%22UserController%20as%20ul%22%20class%3D%22ng-scope%22%26gt%3B
  (Session info: chrome=39.0.2171.71)
  (Driver info: chromedriver=2.14.313457 (3d645c400edf2e2c500566c9aa096063e707c9cf),platform=Linux 3.13.0-40-generic x86_64) (WARNING: The server did not provide any stacktrace information)
Command duration or timeout: 6 milliseconds
Build info: version: '2.45.0', revision: '5017cb8', time: '2015-02-26 23:59:50'
System info: host: 'mark-14', ip: '127.0.1.1', os.name: 'Linux', os.arch: 'amd64', os.version: '3.13.0-40-generic', java.version: '1.8.0_25'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Capabilities [{applicationCacheEnabled=false, rotatable=false, mobileEmulationEnabled=false, chrome={userDataDir=/tmp/.com.google.Chrome.8yKq2F}, takesHeapSnapshot=true, databaseEnabled=false, handlesAlerts=true, version=39.0.2171.71, platform=LINUX, browserConnectionEnabled=false, nativeEvents=true, acceptSslCerts=true, locationContextEnabled=true, webStorageEnabled=true, browserName=chrome, takesScreenshot=true, javascriptEnabled=true, cssSelectorsEnabled=true}]
Session ID: a9c68ee8ad04bd4bb031ea24cb1e6427
    at new bot.Error (/usr/local/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/atoms/error.js:113:18)
    at Object.bot.response.checkResponse (/usr/local/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/atoms/response.js:106:9)
    at /usr/local/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/webdriver/webdriver.js:377:20
    at [object Object].promise.ControlFlow.runInFrame_ (/usr/local/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/webdriver/promise.js:1877:20)
    at [object Object].promise.Callback_.goog.defineClass.notify (/usr/local/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/webdriver/promise.js:2464:25)
    at [object Object].promise.Promise.notify_ (/usr/local/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/webdriver/promise.js:563:12)
    at Array.forEach (native)
    at Object.goog.array.forEach (/usr/local/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/goog/array/array.js:203:43)
    at [object Object].promise.Promise.notifyAll_ (/usr/local/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/webdriver/promise.js:552:16)
    at [object Object].goog.async.run.processWorkQueue [as _onTimeout] (/usr/local/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/goog/async/run.js:125:21)
From: Task: WebDriver.manage().timeouts().setScriptTimeout(11000)
    at [object Object].webdriver.WebDriver.schedule (/usr/local/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/webdriver/webdriver.js:360:15)
    at [object Object].webdriver.WebDriver.Timeouts.setScriptTimeout (/usr/local/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/webdriver/webdriver.js:1367:23)
    at [object Object].Runner.createBrowser (/usr/local/lib/node_modules/protractor/lib/runner.js:181:30)
    at /usr/local/lib/node_modules/protractor/lib/runner.js:257:21
    at _fulfilled (/usr/local/lib/node_modules/protractor/node_modules/q/q.js:797:54)
    at self.promiseDispatch.done (/usr/local/lib/node_modules/protractor/node_modules/q/q.js:826:30)
    at Promise.promise.promiseDispatch (/usr/local/lib/node_modules/protractor/node_modules/q/q.js:759:13)
    at /usr/local/lib/node_modules/protractor/node_modules/q/q.js:573:44
    at flush (/usr/local/lib/node_modules/protractor/node_modules/q/q.js:108:17)
    at process._tickCallback (node.js:415:13)
[launcher] Process exited with error code 1
```

**Comment by bvaughn** [comment]
+1

Ran into this same issue (I think) while trying to write Protractor tests for another project that uses Angular Material. All test assertions pass- but then the test hangs and eventually fails with `Error while waiting for Protractor to sync with the page`

**Comment by marcysutton** [comment]
They were pseudo-code, I don't think those ever worked. PR #37 includes two working tests.

**Comment by marcysutton** [comment]
Closed with 7caacf342a795837bffed7fd39ffdbca8559eecf.
